pages/docs/couple-your-code/couple-your-code-initializing-coupling-data.md

```diff
@@ -5,45 +5,27 @@ keywords: api, adapter, initialization, coupling scheme, restart
 summary: "As default values, preCICE assumes that all coupling variables are zero initially. For fluid-structure interaction, for example, this means that the structure is in its reference state. Sometimes, you want to change this behavior – for instance, you may want to restart your simulation."
 ---
 
-For initializing coupling data, you can add the following **optional** method:
+By default preCICE assumes that all coupling variables are zero initially. If you want to provide non-zero initial values, you can write data before calling `initialize()`. This data will then be used as initial data. To check whether initial data is required, you can use the following function:
 
 ```cpp
-void initializeData();
+bool requiresInitialData()
 ```
 
-Before jumping into the implementation, let's try to clarify how the usual the sequence of events in a serial and in a parallel coupling as studied in [Step 4](couple-your-code-coupling-flow) changes.
-
-TODO: picture
-
-In a serial coupling, only the second participant can send data inside `initializeData()`. In parallel coupling, both participants can initialize data.
-
-The high-level API of preCICE makes it possible to enable this feature at runtime, irrelevant of serial or parallel coupling configuration. To support this feature, we extend our example as follows:
+To support data initialization, we extend our example as follows:
 
 ```cpp
-[...]
-
-const std::string& cowid = precice::constants::actionWriteInitialData();
-
-[...]
-
-int displID = precice.getDataID("Displacements", meshID);
-int forceID = precice.getDataID("Forces", meshID);
-double* forces = new double[vertexSize*dim];
-double* displacements = new double[vertexSize*dim];
```

Comment:
This removal does not really belong to this documentation update, because it is triggered by the update of mesh and data ids. But I did it here anyway.